fieldpath/set.go

```diff
@@ -206,6 +206,32 @@ func (s *Set) WithPrefix(pe PathElement) *Set {
 	return subset
 }
 
+// Leaves returns a set containing only the leaf paths
+// of a set.
+func (s *Set) Leaves() *Set {
```

Comment:
That's probably not optimal, but I would also suggest against spending time optimizing for something we may not need to optimize.

Reply:
On the other hand, the "optimal" solution might be easier to read.

Reply:
Attempted a more performant solution.